pkgs/development/python-modules/nocturne/default.nix

```diff
@@ -1,27 +1,28 @@
 {
+  lib,
   buildPythonPackage,
-  cmake,
   fetchFromGitHub,
-  gtest,
-  hydra-core,
-  lib,
   nlohmann_json,
   pybind11,
-  pyvirtualdisplay,
-  sfml,
   replaceVars,
+  gtest,
+  setuptools,
+  cmake,
+  sfml_2,
+  hydra-core,
+  pyvirtualdisplay,
 }:
 
 buildPythonPackage rec {
   pname = "nocturne";
-  version = "unstable-2022-10-15";
-  format = "setuptools";
+  version = "0-unstable-2024-06-19";
```

Comment:
why `0-unstable-*-*-*`? iiuc `unstable-*-*-*` is the recommended format

Reply:
https://github.com/NixOS/nixpkgs/blob/c1ee2620296430ac1e3ee72583ad0191463a9d60/pkgs/README.md?plain=1#L389

> *- It _must_ start with a digit. This is required > for backwards-compatibility with [how `nix-env` parses derivation names](https://nix.dev/manual/nix/latest/command-ref/nix-env#selectors).*
> 
>   *Example: `"0.3.1rc2"` or `"0-unstable-1970-01-01"`*
> 
> *- If a package is a commit from a repository without a version assigned, then the `version` attribute _should_ be the latest upstream version preceding that commit, followed by `-unstable-` and the date of the (fetched) commit. The date _must_ be in `"YYYY-MM-DD"` format.*